### ISSUE TITLE
Maven checkersplugin result in compile errors despite using "-Awarns" flag

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#177]
```
What steps will reproduce the problem?

1. Add the following configuration to maven project pom.xml

<plugin>
   <groupId>types.checkers</groupId>
      <artifactId>checkersplugin</artifactId>
      <version>0.4</version>
      <executions>
         <execution>
            <phase>process-classes</phase>
            <goals>
               <goal>check</goal>
            </goals>
         </execution>
      </executions>
      <configuration>
      <javacParams>-Awarns -Xmaxwarns 10000</javacParams>
      <processors>
         <processor>checkers.interning.InterningChecker</processor>
         <processor>checkers.nullness.NullnessChecker</processor>
      </processors>
   </configuration>
</plugin>

2. Make sure that there is at least one concern in the Java code that will be 
caught by the Checker Framework.

3. Execute "mvn test"


What is the expected output? 

The project should build successfully with compiler warnings for the problems 
that are detected by the Checker Framework.

What do you see instead?

[INFO] --- checkersplugin:0.4:check (default) @ inject ---
[INFO] Running JSR308 checkers version: 1.4.3
[INFO] Running processor(s): 
checkers.interning.InterningChecker,checkers.nullness.NullnessChecker
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Test Project ....................................... FAILURE [3.018s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[...]
[ERROR] Failed to execute goal types.checkers:checkersplugin:0.4:check 
(default) on project inject: error: Unexpected InvocationTargetException for 
class name checkers.interning.InterningVisitor when invoking the constructor; 
parameter types: [class checkers.interning.InterningChecker, interface 
com.sun.source.tree.CompilationUnitTree] -> [Help 1]


What version of the product are you using? On what operating system?

<groupId>types.checkers</groupId>
<artifactId>checkersplugin</artifactId>
<version>0.4</version>

<groupId>types.checkers</groupId>
<artifactId>checkers-quals</artifactId>
<version>1.4.3</version>

Java: 1.7.0_09
Mac OS X: 10.8.2

Please provide any additional information below.
```

Original issue reported on code.google.com by `mat...@gmail.com` on 19 Nov 2012 at 8:28
